.github/workflows/undeploy-2-start.yml

```diff
@@ -0,0 +1,52 @@
+name: Undeployment Start
+on:
+  workflow_call:
+    inputs:
+      env-name:
+        type: string
+        required: true
+        description: The spack-env-compliant model name to remove
+      deployment-environment:
+        type: string
+        required: true
+        description: The GitHub deployment environment name
+jobs:
+  undeploy-from-environment:
+    name: Undeploy ${{ inputs.env-name }} from ${{ inputs.deployment-environment }}
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.deployment-environment }}
+    steps:
+      - name: Setup SSH
+        id: ssh
+        uses: access-nri/actions/.github/actions/setup-ssh@main
+        with:
+          private-key: ${{ secrets.SSH_KEY }}
+          hosts: |
+            ${{ secrets.HOST }}
+            ${{ secrets.HOST_DATA }}
+
+      - name: Undeploy
+        # ssh into deployment environment, create and activate the env, remove all the unneeded environments
+        id: undeploy
+        run: |
+          ssh ${{ secrets.USER}}@${{ secrets.HOST }} -i ${{ steps.ssh.outputs.private-key-path }} /bin/bash <<'EOT'
+          . ${{ vars.SPACK_CONFIG_LOCATION }}/spack-enable.bash
+          envs=$(find ${{ vars.SPACK_LOCATION }}/var/spack/environments -type d -name '${{ inputs.env-name }}' -printf '%f ')
+
+          for env in $envs; do
+            spack env activate $env
+            spack uninstall --remove --dependents --yes-to-all --all
+            spack env deactivate $env
+            spack env rm $env --yes-to-all
+          done
+          spack gc --yes-to-all
```

Comment:
No, it doesn't seem to. We use it to clean up packages that might not have been uninstalled by the repeated `spack uninstall` calls because they were still dependent on other environments that hadn't been removed yet.